ballerina/client.bal

```diff
@@ -256,6 +256,78 @@ public isolated client class Client {
+    # + action -   The HTTP method or the HTTP method with respective headers(objectCreationHeaders or objectRetrievalHeaders)
+    # + expires - The time period for which the presigned URL is valid, in seconds  
+    # + partNo - The part number of the object, when uploading multipart objects  
+    # + uploadId - The upload ID of the multipart upload
+    # + return - If successful, a presigned URL, else an error
+    @display {label: "Create Presigned URL"}
+    remote isolated function createPresignedUrl(
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "Object Name"} string objectName,
+            @display {label: "HTTP Method and Optional Headers"} GET|PUT|GetHeaders|PutHeaders action,
+            @display {label: "Expiration Time"} int expires = 1800,
+            @display {label: "Part Number"} int? partNo = (),
+            @display {label: "Upload ID"} string? uploadId = ())
+        returns string|error {
+        
+        if (expires < 0) {
+            return error(EXPIRATION_TIME_ERROR_MSG);
+        }
+        if objectName == "" {
+            return error(OBJECT_NAME_ERROR_MSG);
+        }
+        if bucketName == "" {
+            return error(BUCKET_NAME_ERROR_MSG);
+        }
+
+        [string, string] [amzDateStr, shortDateStr] = check generateDateString();
+
+        map<string> requestHeaders = {
+            [HOST] : string `${self.amazonHost}`
+        };
+    
+        GET|PUT httpMethod = action is GET|PUT ? action : action.method;
+        
+        if action is PutHeaders{
+            populateCreateObjectHeaders(requestHeaders, action.headers);
+        } else if action is GetHeaders {
+            populateGetObjectHeaders(requestHeaders, action.headers);
+        }
+
+        [string, string] [canonicalHeaders, signedHeaders] = generateCanonicalHeaders(requestHeaders, ());
+
+        map<string> queryParams = {
+            [X_AMZ_ALGORITHM] : AWS4_HMAC_SHA256,
+            [X_AMZ_CREDENTIAL] : string `${self.accessKeyId}/${shortDateStr}/${self.region}/${SERVICE_NAME}/${
+                TERMINATION_STRING}`,
+            [X_AMZ_DATE] : amzDateStr,
+            [X_AMZ_EXPIRES] : expires.toString(),
+            [X_AMZ_SIGNED_HEADERS] : signedHeaders
+        };
+
+        string|error canonicalQuery = generateCanonicalQueryString(queryParams);
+        if canonicalQuery is error {
+            return error(CANONICAL_QUERY_STRING_GENERATION_ERROR_MSG, canonicalQuery);
+        } 
+
+        if partNo != () && uploadId != () && httpMethod == PUT {
+            canonicalQuery = string `${canonicalQuery}&partNumber=${partNo}&uploadId=${uploadId}`;
+        }
+        canonicalQuery = re `/`.replaceAll(check canonicalQuery, "%2F");
+        string canonicalRequest = string `${httpMethod}${"\n"}/${bucketName}${string `/${objectName}`}${"\n"}${check canonicalQuery}${
+            "\n"}${canonicalHeaders}${"\n"}${signedHeaders}${"\n"}${UNSIGNED_PAYLOAD}`;
```

Comment:
If you use a different variable for canonicalQuery after L317 you can avoid using check, since the value is string.

Reply:
Wherever `check canonicalQuery` is used, L328 too.

Reply:
so canonicaQuery should be assigned to another string variable?